src/Gui/CallTips.cpp

```diff
@@ -472,20 +472,7 @@ void CallTipsList::showTips(const QString& line)
     static QPixmap property_icon = BitmapFactory().pixmap("ClassBrowser/property.svg");
 
     // object is in error state
-    static const char * const forbidden_xpm[]={
-            "8 8 3 1",
-            ". c None",
-            "# c #ff0000",
-            "a c #ffffff",
-            "..####..",
-            ".######.",
-            "########",
-            "#aaaaaa#",
-            "#aaaaaa#",
-            "########",
-            ".######.",
-            "..####.."};
-    static QPixmap forbidden_icon(forbidden_xpm);
+    static QPixmap forbidden_icon(Gui::BitmapFactory().pixmapFromSvg("forbidden", QSize(8, 8)));

```

Comment:
Here you can use a size of 24x24 or 32x32 as otherwise the overlay icon will be too small.

Reply:
but the icon will be displayed too large then. When I scale it down after a higher resolution with `.scaled` 
Do I need to call the get dpr every time then, if https://github.com/FreeCAD/FreeCAD/pull/13176 is in?

Reply:
When using 8x8 then the overlay icon is extremely tiny. With 24x24 or 32x32 the size is sensible for me.

Reply:
This is how it looks with 8x8 (XPM and SVG):
![forbidden_8x8](https://github.com/FreeCAD/FreeCAD/assets/7101741/d98836a2-f021-4e14-ab44-15fbc643e1e5)
and this with 24x24:
![forbidden_24x24](https://github.com/FreeCAD/FreeCAD/assets/7101741/8256b5db-821e-45db-9b30-31fd86507627)

Reply:
On Windows it looks like the bottom one with the 10x10 or 8x8

Reply:
When I check the size of type_module_icon, type_module_icon, ... then it says it's 64x64. So, the 8x8 is only an eighth of it and thus the overlay icons are so tiny. When setting the 64 x 64 icon to the list item it's scaled down to the used size.

I wonder  what are the sizes of the above icons on Windows. Are they smaller than 64x64?

Reply:
I have tested it on Windows now and there I have the same problem as under Linux. The size of the type_module_icon is 64x64. I can't imagine that on your system it's of the same size.

Reply:
@wwmayer on 
<details><summary>Windows 11 with this PR as is</summary>
<p>

```
OS: Windows 11 build 22631
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36391 (Git)
Build type: Release
Branch: main
Hash: 52fffaf50ef8facea2ecfde23d6f1a7d3ad3d80c
Python 3.11.8, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * 3DfindIT 1.2.0
  * BIM 2021.12.0
  * CfdOF 1.25.3
  * CurvedShapes 1.0.8
  * Curves 0.6.29
  * Defeaturing 1.2.2
  * fasteners 0.5.15
  * FEMbyGEN 2.1.0
  * freecad.gears 1.2.0
  * freecad_metal_workbench 0.0.1
  * OpenDark 2023.12.17
  * sheetmetal 0.4.6
  * sheetmetal.backup1710961794.9712274 0.4.3 (Disabled)

``` 

</p>
</details> 

using 10x10 px: 
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/c2b61574-6871-4257-b17d-8a74220116b0)

using 64x64 px:
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/046d64bf-a5f3-4a90-9ef4-90d271ea0f0c)

Definitely not what I want, Icon is bigger than the icon below.

Reply:
You have tested the wrong stuff. I was only talking about the icons of the Calltips window. You have to check the method CallTipsList::showTips() where you check the size of the created pixmap. 

Add this line after the creation of the first pixmap:
`Base::Console().Message("(%d, %d)\n", type_module_icon.width(), type_module_icon.height());`
Start FreeCAD and enter "App." into the Python console. Check the report window.

Reply:
`(64, 64)` the same. but it's so pixelated, you can see in the above screenshots in comparison with the underlying icn of the feature.

Reply:
@wwmayer so should I adjust some code to get it without pixelation or save this for another PR as its currently also pixelated?

Reply:
I think the PR can be merged now. To me the new SVG overlay icons look much better than the XPM variants.

Now, what are the remaining problems and which exact icons don't look good? The only remaining issue for me is that the overlay icons for _forbidden_ of the calltips or so tiny.

Reply:
For me, on a HiDPI display, all overlay icons look pixelated, more than the underlying icon they are merged onto.
You can see it in this comment: https://github.com/FreeCAD/FreeCAD/pull/13110#discussion_r1543351652

Reply:
> all overlay icons look pixelated,

Is it as bad as with the XPM overlay icons or is it somewhat better?

Reply:
I'd say equal to before.
Left is before, right is with this PR:
![grafik](https://github.com/FreeCAD/FreeCAD/assets/6246609/cbc7bd4e-0ae9-49de-9459-a2ab6e007c68)

If you compare it with the Body or Boy icon below, it is more pixelated than the underlying ones.

Reply:
I would say the right one is slightly better because you can clearly see some details.

Can you figure out what's the size of the tree icons? To do so add this line to the method `TreeWidget::iconSize()`:
`Base::Console().Message("Tree icon size: %d\n", defaultSize);`

On a normal screen the tree icons have a size of 16x16 so that an overlay icon with 8x8 should occupy 50% of the width & height. According to your screenshot this is clearly less -- maybe a third or fourth.

IMO using a hard-coded size of 8x8 is a problem because depending on system settings or style(sheet) the tree icons may have a different size than 16x16 and the overlay icon should grow with the same ratio.

Reply:
`Tree icon size: 16`

Reply:
But if your tree icon has a size of 16x16 then the overlay icon should occupy 50% of the width/height which is clearly not the case.

So, at some point an icon of a bigger size is created and later scaled down while the overlay icon was already created with the size of 8x8.

For my calltips issue I have changed the code to create the overlay icon to
`static QPixmap forbidden_icon(Gui::BitmapFactory().pixmapFromSvg("forbidden", property_icon.size() / 4));`
and this now looks as expected.

Reply:
it looks better when I create the overlay icon at 20x20 and add .scaled to 10x10.

Reply:
You can try the overloaded method of pixmapFromSvg() that accepts the DPR as third argument.

Reply:
But then I need to call the function to get the current DPR everywhere an overlay icon is specified.

Reply:
You can at least use it in DocumentObjectItem::testStatus() where you can access the tree widget.

Reply:
but overlay icons are set from almost all WBs. Could this be integrated in `pixmapFromSvg()` instead? If no DPR is given, calculate it there?

Reply:
No, as this only makes no sense. On systems with several monitors it's possible that widgets have different DPR and pixmapFromSvg() doesn't know anything about the context a pixmap is used.

Reply:
how is it working with toolbar icons? On my second monitor DPI they also look good. dragging them from one to another monitor updates them.

Reply:
I don't know. I guess Qt does it automatically.